sycl/source/kernel_bundle.cpp

```diff
@@ -290,12 +291,59 @@ std::vector<kernel_id> get_kernel_ids() {
 }
 
 bool is_compatible(const std::vector<kernel_id> &KernelIDs, const device &Dev) {
-  std::set<detail::RTDeviceBinaryImage *> BinImages =
-      detail::ProgramManager::getInstance().getRawDeviceImages(KernelIDs);
-  return std::all_of(BinImages.begin(), BinImages.end(),
-                     [&Dev](const detail::RTDeviceBinaryImage *Img) {
-                       return doesDevSupportDeviceRequirements(Dev, *Img);
-                     });
+  if (KernelIDs.empty())
+    return false;
```

Comment:
Why should `is_compatible` return false if `KernelIDs` is empty?

Reply:
Spec says: `Returns: true if all of the kernels identified by kernelIds are compatible with the device dev.` I guess for this case since we don't have kernels at all, we shouldn't return true here. Do you have another opinion?

Reply:
How do we handle empty list of kernels in `has_kernel_bundle`?  `std::all_of` returns `true` for an empty array, so, from this point, returning `true` for empty list of kernels would make sense.

The standard does not explicitly cover this case, and SYCL CTS [explicitly skips the cases of empty vectors](https://github.com/KhronosGroup/SYCL-CTS/blob/e9ffbe9a2cad6b8a31c5bab58d05d836a2338df9/tests/kernel_bundle/sycl_is_compatible.cpp). But their use of `CHECK(sycl::is_compatible(builtinKernelIds, device))` suggests that returning `true` for an empty array would be more consistent with how this function is expected to be used.

Reply:
Okay, I clarified this point, it should return true - https://github.com/intel/llvm/pull/9903